sql/moz-fx-data-shared-prod/cloudflare_derived/os_usage_v1/metadata.yaml

```diff
@@ -9,7 +9,10 @@ labels:
 scheduling:
   dag_name: bqetl_cloudflare_os_market_share
   date_partition_parameter: dte
-  arguments: ["--date", "{{ds}}", "--cloudflare_api_token", "{{ var.value.cloudflare_auth_token}}"]
+  arguments: ["--date", "{{ds}}"]
+  secrets:
+  - deploy_target: CLOUDFLARE_AUTH_TOKEN
+    key: CLOUDFLARE_AUTH_TOKEN
```

Comment:
In this case there are three different DAGs wanting to use the same secret.  Should we have three DAG-specific copies of the secret, or stick with a single secret?

Reply:
Also I'm curious, why are these three Cloudflare ETLs in separate DAGs (each of which only have a single task)?

I see they're scheduled six hours apart, maybe to spread out the calls to the Cloudflare API and avoid triggering quota limits?  But they don't seem to be super intensive (the longest one has been taking ~1 hour, and the two others have been taking just a few minutes).

IMO it'd make sense to put these ETLs in a single DAG since they're closely related, unless there's a specific technical reason not to.

Reply:
Hey Sean - I did it to avoid rate-limiting issues (it's an open source API and only lets you submit so many requests in a 24 hour period so I split into 3 DAGs so I could put each one 8 hours apart).  The browser one in particular makes a lot of calls at once.  I'm going to put them into 3 separate keys, 1 for each DAG (I just think it's easier to put them into 3 keys). That way it'll be very clear which key is for which DAG.